lib/Algo/Algorithm.h

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+#include <tuple> 
+#include "../board_implementation/CFBoard.h"
+
+class Algorithm {

```

Comment:
This name is too confusing (compared to `<algorithm>` of C++). At least if you want this name, wrap it inside some namespace first. Otherwise, I'd suggest `ChessEngine` or similar.

Reply:
Fixed, thanks for the comment 👍